x-pack/plugins/security_solution/server/lib/detection_engine/signals/types.ts

```diff
@@ -342,3 +342,8 @@ export type EventGroupingMultiBucketAggregationResult = ESSearchResponse<
     };
   }
 >;
+
+export interface OverrideBodyQuery {
+  _source: estypes.SearchSourceConfig;
+  fields: estypes.Fields;
```

Comment:
Good suggestion!

I changed those fields to optional.

I was trying to change `OverrideBodyQuery ` to `estypes.SearchRequest['body']`, but it wasn't working, because looks like 
`buildEventsSearchQuery` return compatible but not the exact type.

Later in a single search after we cast this query to `estypes.SearchRequest`

So if somebody need to add new fields, it shouldn't be a problem to add it into `OverrideBodyQuery `

Reply:
thanks for looking into it.
I think it's good to go. 
With added comment it should not be an issue to add new fields if needed